doc/source/configuration/ci-cd.rst

```diff
@@ -0,0 +1,179 @@
+Deployment of private runners is supported by all major providers with the use of community developed Ansible roles.
+Finally, due to the requirement that we support various different platforms tooling in the form of `Kayobe automation <https://github.com/stackhpc/kayobe-automation/>`__ was developed.
+This tooling is not tied to any single CI/CD platform as all tasks are a series of shell script and Ansible playbooks which are designed to run in a purpose build kayobe container.
+This is complemented by the use of an Ansible collection known as `stackhpc.kayobe_workflows <https://github.com/stackhpc/ansible-collection-kayobe-workflows/>`__ which aims to provide users with a quick and easy way of customising all workflows to fit within a customer's cloud.
+
+Currently we support the creation and deployment of workflows for GitHub with Gitlab support being actively worked upon. 
+
+Kayobe Automation
+-----------------
+
+Workflows
+---------
+
+`Kayobe automation <https://github.com/stackhpc/kayobe-automation/>`__ is a collection of scripts and tools that automate kayobe operations.
+It is deployed and controlled by CI/CD platforms such as GitHub actions and GitLab pipelines.
+Kayobe automation provides users with an easy process to perform tasks such as: overcloud service deploy, config-diff, tempest testing, and many more.
+With it being integrated into platforms such as GitHub or GitLab it builds a close relationship between the contents of the deployments kayobe configuration and what is currently deployed.
+This is because operations such as opening a pull request will trigger a config diff to be generated providing insight on what impact it might have on services or a tempest test that could be scheduled to run daily providing knowledge of faults earlier than before.
+
+Kayobe automation has been designed to be independent of any CI/CD platform with all tasks running inside of a purpose built Kayobe container.
+However, platform specific workflows need to be deployed to bridge the gap between the contents of Kayobe Automation and these CI/CD platforms.
+Workflows are templated for each Kayobe configuration repository, ensuring appropriate workflow input parameters are defined, and any necessary customisations can be applied.
+The templating of workflows is offered through the `stackhpc.kayobe_workflows <https://github.com/stackhpc/ansible-collection-kayobe-workflows/>`__ collection which currently supports GitHub workflows.
+
+Runners
+-------
+
+Runners are purpose built services tied to a particular service vendor such as GitHub Actions or GitLab CI.
+These services will listen for jobs which have been tagged appropriately and dispatched to these specific runners.
+The runners will need to be deployed using existing roles and playbooks whereby the binary/package is downloaded and registered using a special token.
```

Comment:
Could mention here that runners generally need to be deployed for each kayobe environment.

Reply:
Explained it might be possible to use a shared host however this might not be possible and will require deployment of runners within all environments